### PR TITLE
Lodash: split up monolith file and use per file per method.

### PR DESCRIPTION
**Pull request by DomiR**
As discussed in #23047.

This is quite a heavy change but I think it could help to improve the quality even more.

I chose to use the structure from the lodash documentation website (I wrote a helper script that made the structure and split up the index.d.ts automatically). There are some stub files that could be deleted at some point. I manually checked that all method files contained content and touched up some of them.

This is all I can do with two hours. I thought I would leave the test for now, so the split can be verified by others easily and maybe my changes can even be merged first. Next weekend I will try to move the tests into the appropriate files, but only if this gets approved.

@aj-r Would love to get some feedback on this 😅 

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:
If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

File: types/lodash/lodash-tests.ts [hunk 1]
```diff
@@ -8098,7 +8098,7 @@ namespace TestCloneWith {
 
 // _.conforms
 namespace TestConforms {
-    let result: boolean = _.conforms({foo: (v: string) => false})({foo: "foo"});
+    let result: boolean = _.conforms({foo: (v: {}) => false})({foo: "foo"});
```
Comment (aj-r) [inline types/lodash/lodash-tests.ts]: What's the purpose of this change? Shouldn't `v` be a string?

File: types/lodash/index.d.ts [hunk 2]
```diff
@@ -8,17381 +8,367 @@
...
-- [ ] _.cond
-- [ ] _.nthArg
-- [ ] _.over
-- [ ] _.overEvery
-- [ ] _.overSome
-- [ ] _.rangeRight
-
-- [ ] _.next
-*/
+// common
```
Comment (aj-r) [inline types/lodash/index.d.ts]: Seems like Travis is failing to parse the header. Maybe it's because of this comment? Not sure. Have you run `npm run lint lodash`?

**Review by aj-r — CHANGES_REQUESTED** [review]
Are all of the test files really empty? Or is github just not showing the contents? You should either move the contents of `lodash-tests.ts` into each individual test file, or delete all of the test files. I prefer the former. If you go that route, I would delete `lodash-tests.ts` and instead reference each of the test files in `tsconfig.json`.

I'm assuming that you didn't make any functional changes, and only copy/pasted the code into different files? If you ever did anything other than copy/pasting, I'd appreciate it if you pointed it out :)

**Comment by DomiR** [comment]
I did not change any functionality. I just removed some faulty comments.

As you already saw it is quite hard to check if some functionality was changed, dropped or added, so I thought I would leave the test unchanged first. Next weekend I can look into splitting up the tests as well, so it's definitely planned but this could already be merged in, as soon as the build runs through.

The test will fail with the message if I leave the string there: 
```
Argument of type '{ foo: (v: string) => boolean; }' is not assignable to parameter of type 'ConformsPredicateObject<{ foo: any; }>'.
  Types of property 'foo' are incompatible.
    Type '(v: string) => boolean' is not assignable to type '((val: {}) => boolean) | undefined'.
      Type '(v: string) => boolean' is not assignable to type '(val: {}) => boolean'.
        Types of parameters 'v' and 'val' are incompatible.
          Type '{}' is not assignable to type 'string'.'
at: '8101,38'
source: 'ts'
code: '2345'
```

File: types/lodash/lodash-tests.ts (hunk 1 above)
Comment (DomiR) [reply types/lodash/lodash-tests.ts]: The test will fail with the message if I leave the string there: 
```
Argument of type '{ foo: (v: string) => boolean; }' is not assignable to parameter of type 'ConformsPredicateObject<{ foo: any; }>'.
  Types of property 'foo' are incompatible.
    Type '(v: string) => boolean' is not assignable to type '((val: {}) => boolean) | undefined'.
      Type '(v: string) => boolean' is not assignable to type '(val: {}) => boolean'.
        Types of parameters 'v' and 'val' are incompatible.
          Type '{}' is not assignable to type 'string'.'
at: '8101,38'
source: 'ts'
code: '2345'
```

File: types/lodash/index.d.ts (hunk 2 above)
Comment (DomiR) [reply types/lodash/index.d.ts]: Found a missing `>`... my bad.

File: types/lodash/lodash-tests.ts (hunk 1 above)
Comment (aj-r) [reply types/lodash/lodash-tests.ts]: Strange because this test passed before. Look into why this test is failing, and see if you can make it pass with `string` instead of `{}`. Otherwise you're introducing a breaking change!

**Review by aj-r — CHANGES_REQUESTED** [review]
See if you can make that one test pass with `string` instead of `{}`. Somehow the types must have changed - maybe a copy pasta error?

**Comment by DomiR** [comment]
Well `tsc(2.6.2)` will not throw, but vscode with `2.7.0-insiders.20180117` will show me the described error. I reverted my changes, but this will maybe not pass in the next `tsc` version...

**Review by aj-r — APPROVED** [review]
Looks good.

That "failing" test either indicates a bug in the version of typescript you're using (`2.7.0-insiders.20180117`) or a bug in lodash. Either way the test itself looks correct. If the Travis CI build starts failing when typescript 2.7 is released (or near release), we'll fix it at that time.

File: types/lodash/util/uniqueId.d.ts [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+         *
+         * @param prefix The value to prefix the ID with.
+         * @return Returns the unique ID.
+         */
+        uniqueId(prefix?: string): string;
+    }
+
+    interface LoDashImplicitWrapper<TValue> {
+        /**
+         * @see _.uniqueId
```
Comment (andy-ms) [inline types/lodash/util/uniqueId.d.ts]: Maybe these comments are unnecessary now? We're *at* `_.uniqueId`.

Comment (thorn0) [reply types/lodash/util/uniqueId.d.ts]: These comments were apparently made to write the description only once in the first overload of the method in the hope that IDEs will be smart enough to parse these `@see` annotations and inherit JSDoc somehow. See Microsoft/TypeScript/issues/407
